mailpoet/mailpoet
# Refactor Menu [MAILPOET-2200]

I've added a new namespace `MailPoet\AdminPages` and refactored all methods used for fetching data for templates to extra classes and injected container so that a class for a particular page is instantiated when it is needed for rendering.

## QA Notes
I would be good to go thru all MailPoet admin pages and check that all renders correctly.
Here are all pages which are handled by Menu class:

mailpoet/lib/AdminPages/Pages/ExperimentalFeatures.php
mailpoet/lib/AdminPages/Pages/FormEditor.php
mailpoet/lib/AdminPages/Pages/Forms.php
mailpoet/lib/AdminPages/Pages/Help.php
mailpoet/lib/AdminPages/Pages/MP2Migration.php
mailpoet/lib/AdminPages/Pages/NewsletterEditor.php
mailpoet/lib/AdminPages/Pages/Newsletters.php
mailpoet/lib/AdminPages/Pages/Premium.php
mailpoet/lib/AdminPages/Pages/RevenueTrackingPermission.php
mailpoet/lib/AdminPages/Pages/Segments.php
mailpoet/lib/AdminPages/Pages/Settings.php
mailpoet/lib/AdminPages/Pages/Subscribers.php
mailpoet/lib/AdminPages/Pages/SubscribersAPIKeyInvalid.php
mailpoet/lib/AdminPages/Pages/SubscribersExport.php
mailpoet/lib/AdminPages/Pages/SubscribersImport.php
mailpoet/lib/AdminPages/Pages/SubscribersLimitExceeded.php
mailpoet/lib/AdminPages/Pages/Update.php
mailpoet/lib/AdminPages/Pages/WelcomeWizard.php
mailpoet/lib/AdminPages/Pages/WooCommerceListImport.php
